packages/grid/_modules_/grid/hooks/utils/useResizeContainer.ts

```diff
@@ -38,7 +38,7 @@ export function useResizeContainer(apiRef): (size: ElementSize) => void {
     },
     [gridLogger, apiRef],
   );
-  const debouncedOnResize = React.useMemo(() => debounce(onResize, 100), [onResize]) as any;
+  const debouncedOnResize = React.useMemo(() => debounce(onResize, 10), [onResize]) as any;
```

Comment:
So it uses the default value we use to denounce resize events:

https://github.com/mui-org/material-ui/blob/a20f0a826acc1c1fc53ca8f3b852c63f3fa291bd/packages/material-ui/src/utils/debounce.js#L1

? 

```suggestion
  const debouncedOnResize = React.useMemo(() => debounce(onResize), [onResize]);
```

Reply:
in 166ms we have time for at least 3 renderings. So I would prefer to just add a bit of delay to process the size, and trigger the resize so it can be achieved in  60ms

Reply:
166ms was picked per https://www.nngroup.com/articles/response-times-3-important-limits/, to be a bit above 100ms and to be also 10 frames at 60 Hz.

When you resize your screen, it might take you 2s, during these 2s, you fire a resize event at, say 15 Hz, it's above the denounce value at 6 Hz. So the grid waits until the end of the session to rerender. If we pick 10ms, that's 100 Hz, so the grid would rerender 200 times, instead of 1 time in my example. Do we need this behavior?

Reply:
Maybe 1/60*5 would be perfect? 83ms, 5 screens, and leave space to be below the 100ms limit?

Reply:
Up to you. Just need to update the sleep in the test accordingly

Reply:
I would recommand this approach:

- We use the default value
- We update the test from 100 to 166
- We change the core from 166 to 83, we will leverage it once we go to v5, until then, it's a small detail.

Reply:
![image](https://user-images.githubusercontent.com/936978/98017185-318b2280-1dff-11eb-848c-4c3d4a3469d5.png)

TBH I would leave 10 or 30 - 50 max, I think for a resize event > 100 is far too long. Also not setting the parameter, allows the lib to create a bug if someone changes it.

Reply:
I think that would help to agree on the use case we are considering. What' the use case?

Mine was 1. an action that changes the size of the grid, like clicking a button that opens a panel, and 2. a user manually resizing its window to fit something. 83 is a compromise to be hight enough to not render hundreds of times during 2. and small enough to not be perceptible for 1.

What's the link with the screenshot?

Reply:
all the tests failing with 166

Reply:
See https://github.com/mui-org/material-ui-x/pull/532/commits/58ee78c7aa55a542dccd7714a6409788fddbfb60